src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.cs

```diff
@@ -376,13 +367,25 @@ SynthesizedClosureEnvironment MakeFrame(Analysis.Scope scope, bool isStruct)
                     containingMethod = _substitutedSourceMethod;
                 }
 
-                return new SynthesizedClosureEnvironment(
+                var synthesizedEnv = new SynthesizedClosureEnvironment(
                     _topLevelMethod,
                     containingMethod,
-                    isStruct,
+                    env.IsStruct,
                     syntax,
                     methodId,
                     closureId);
+
+                foreach (var captured in env.CapturedVariables)
+                {
+                    Debug.Assert(!proxies.ContainsKey(captured));
+
+                    var hoistedField = LambdaCapturedVariable.Create(synthesizedEnv, captured, ref _synthesizedFieldNameIdDispenser);
+                    proxies.Add(captured, new CapturedToFrameSymbolReplacement(hoistedField, isReusable: false));
+                    synthesizedEnv.AddHoistedField(hoistedField);
```

Comment:
>synthesizedEnv.AddHoistedField(hoistedField); [](start = 20, length = 45)

Seems like all the hoisted fields are known at construction time of SynthesizedClosureEnvironment. Can we pass them in, instead of using a builder in the environment?